tests/regressions/return_void.c3

```diff
@@ -0,0 +1,7 @@
+function main : () -> int = {
+    return;
+}
+
+/// @COMPILE; EXPECT ERR
+/// File '*.c3', line 2, in 'function main: () -> int'
+/// Error in return statement: type 'int' does not match expected type 'void'
```

Comment:
I've defo encountered this before. Could have sworn I opened a bug... But I guess not. Thanks for fixing it :heart: